### ISSUE TITLE
About sds loss

### ISSUE THREAD

**Issue by OrangeSodahub** [JunzheJosephZhu/HiFA#8]
Hi, after reading your paper. I just want to confirm that: using a sds method is equal to use the outputs from diffusion model (e.g. latents, rgb images) to supervise a standard nerf model? Is that correct?

**Comment by HiFA-team** [comment]
true. diffusion model outputs noise, so you have to basically subtract the
noise from the noisy image to get the "outputs" you're talking about

**Comment by OrangeSodahub** [comment]
Yeah I know. But I got a little bit confused. In this case, how diffferent between "sds" and train a standard nerf model? The rgb images from diffusion at different times for the same prompt are different, they aren't consistent, how can they fit a nerf model and converge?

**Comment by HiFA-team** [comment]
its the same training process. It probably converges since it's optimizing
for a KL objective, if you look into the math at DreamFusion's appendix.

**Comment by OrangeSodahub** [comment]
@HiFA-team Thanks. And I have another question, can I modify the loss function to just L1 Loss here? And can I remove the `w` term for simplification?

https://github.com/JunzheJosephZhu/HiFA/blob/d0213bcbdfcbe6181ed1960f6b0a8b72dd7909a4/nerf/sd.py#L293

**Comment by HiFA-team** [comment]
you can try it, removing w might cause an imbalance with other loss terms.
I tried L1, it didnt really work that well

**Comment by OrangeSodahub** [comment]
And the `w` term you used in your paper is the `constant`  one?

**Comment by HiFA-team** [comment]
I think it was. The default option should be constant.

**Comment by OrangeSodahub** [comment]
Thanks, and I noticed that you also write about perceptual loss here:

https://github.com/JunzheJosephZhu/HiFA/blob/d0213bcbdfcbe6181ed1960f6b0a8b72dd7909a4/nerf/utils.py#L506-L532

Did you show the results of this part in your paper? How is the reuslt of "not use_sds" ?

**Comment by JunzheJosephZhu** [comment]
That's for the image-guided generation part, assuming there's ground truth generated by syncdremaer

**Comment by OrangeSodahub** [comment]
Thanks, and I found the fact showed in Fig.5 of your paper is interesting, and you said "However, relying solely on the image-space loss LSDS-Image results in color bias issues, regardless of the guidance scale used.", I wonder do you have any options about why using soly image loss is bad?

![image](https://github.com/JunzheJosephZhu/HiFA/assets/54439582/a6db46ea-954f-4779-aefd-62508abf8803)

**Comment by HiFA-team** [comment]
I think there's some sort of nonlinear relationship between latent space
and rgb space, and the diffusion model is trained on the latent space. I
don't really have a good idea though, CFG=100 pretty much breaks all the
usual theoretical intuitions one would have for a diffusion model...

**Comment by OrangeSodahub** [comment]
Thanks, I'm also quite confused about this part:

https://github.com/JunzheJosephZhu/HiFA/blob/1bbe86135f960f4f99ab1f4c294bb5f4151da273/nerf/sd.py#L350-L355

Is there any reference e.g. math equations for the calculation? Why `* (256 * 256)`

**Comment by OrangeSodahub** [comment]
On the other hand, can I say: if I use L1 loss instead of L2 loss, then sds method actually not being used (and the knowledge of diffusion will not be used), here is reason:

```
# sds
∂loss/∂θ = w(ε - ε0) * ∂z/∂θ = ∂loss/∂z * ∂z/∂θ
# your method
∂loss/∂θ = w * αt/σt *(z - z0) * ∂z/∂θ
```

Validation:
```
# 1
loss = w * ||z - z0||^2
∂loss/∂θ = ∂(w * ||z - z0||^2)/∂θ = 2w * (z - z0) * ∂z/∂θ
# 2
loss = w * ||z - z0||
∂loss/∂θ = ∂(w * ||z - z0||)/∂θ = 2w * (±1) * ∂z/∂θ
```

**Comment by JunzheJosephZhu** [comment]
I would say not as much knowledge is used, or the gradient is quantized to {-1, 1}
